duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
-            sr.total_xsSyst_cv_wgt *= it.CV_response;
+  void CAFMaker::FillRecoInfo(caf::SRCommonRecoBranch &recoBranch, const art::Event &evt) const {
+    SRInteractionBranch &ixn = recoBranch.ixn;
+
+    //Only filling with Pandora Reco for the moment
+    std::vector<SRInteraction> &pandora = ixn.pandora;
+    
+    lar_pandora::PFParticleVector particleVector;
+    lar_pandora::LArPandoraHelper::CollectPFParticles(evt, fPandoraNuVertexModuleLabel, particleVector);
+    lar_pandora::VertexVector vertexVector;
+    lar_pandora::PFParticlesToVertices particlesToVertices;
+    lar_pandora::LArPandoraHelper::CollectVertices(evt, fPandoraNuVertexModuleLabel, vertexVector, particlesToVertices);
+
+    for (unsigned int n = 0; n < particleVector.size(); ++n) {
+      const art::Ptr<recob::PFParticle> particle = particleVector.at(n);
+      if(particle->IsPrimary()){
+        SRInteraction reco;
+
+        //Retrieving the reco vertex
+        lar_pandora::PFParticlesToVertices::const_iterator vIter = particlesToVertices.find(particle);
+        if (particlesToVertices.end() != vIter) {
+          const lar_pandora::VertexVector &vertexVector = vIter->second;
+          if (vertexVector.size() == 1) {
+            const art::Ptr<recob::Vertex> vertex = *(vertexVector.begin());
+            double xyz[3] = {0.0, 0.0, 0.0} ;
+            vertex->XYZ(xyz);
+            reco.vtx = SRVector3D(xyz[0], xyz[1], xyz[2]);
+            // fData->nuvtxpdg[iv] = particle->PdgCode(); TODO: Reuse this elsewhere, add a branch in SRNeutrinoHypothesisBranch for it
           }
         }
+
+        SRDirectionBranch &dir = reco.dir;
+        FillDirectionInfo(dir, evt);
+
+
+        //Neutrino flavours hypotheses
+        SRNeutrinoHypothesisBranch &nuhyp = reco.nuhyp;
+        //Filling only CVN at the moment. Filling the same info for all the particles -> CVN applies to the whole event as far as I know
+        FillCVNInfo(nuhyp.cvn, evt);
+
+        //Neutrino energy hypothese
+        SRNeutrinoEnergyBranch &Enu = reco.Enu;
+        FillEnergyInfo(Enu, evt);
+
+        //List of reconstructed particles
+        SRRecoParticlesBranch &part = reco.part;
+        FillRecoParticlesInfo(part, evt);
+
+        reco.truth = {0}; //Assuming a single TrueInteraction for now
+        //TODO, not sure of what to put there..
+        // std::vector<float>   truthOverlap;              ///< Fractional overlap between this reco interaction and each true interaction
+
+        pandora.emplace_back(reco);
       }
-    } // loop through MC truth i
+    }
+
+    ixn.npandora = pandora.size();
+    ixn.ndlp = ixn.dlp.size();
   }
 
+
   //------------------------------------------------------------------------------
+ 
+ 
   void CAFMaker::beginSubRun(const art::SubRun& sr)
   {
-    auto pots = sr.getHandle< sumdata::POTSummary >("generator");
-    if( pots ) meta_pot += pots->totpot;
+    art::Handle<sumdata::POTSummary> pots = sr.getHandle<sumdata::POTSummary>("generator");
```

Comment:
Actually, could you change this so that the module label isn't hardcoded and instead passed in as a fcl param